File: python/resistor-color-duo/resistor_color_duo.py
def value(colors):
  value = ''
  for color in colors:
    value += str(valor().index(color))
  return int(value)

def valor():
  return [
      'black',
      'brown',
      'red',
      'orange',
      'yellow',
      'green',
      'blue',
      'violet',
      'grey',
      'white'
  ]

File: typescript/resistor-color-duo/resistor-color-duo.ts
interface Color {
  [key: string]: number;
}

const colors: Color = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};

export function decodedValue(values: string[]): number {
  if (values.length < 2) {
    throw `At least two colors are expected.`;
  }

  const [colorA, colorB] = values;

  return parseInt(`${colors[colorA]}${colors[colorB]}`);
}
